public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -100,35 +102,29 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
     });
   }
 
-  private _updateDirty = debounce(() => {
-    const diff = jsonDiff(vizPanelToPanel(this.state.sourcePanel.resolve()), vizPanelToPanel(this.state.panel));
-    const diffCount = Object.values(diff).reduce((acc, cur) => acc + cur.length, 0);
-    this.setState({ isDirty: diffCount > 0 });
+  private _onActivate() {
+    this.loadDataSource();
+    const changesSub = this.subscribeToEvent(SceneObjectStateChangedEvent, this._handleStateChange);
+
+    return () => {
+      changesSub.unsubscribe();
+    };
+  }
+
+  private _detectPanelModelChanges = debounce(() => {
+    const { hasChanges } = getPanelChanges(
+      vizPanelToPanel(this.state.sourcePanel.resolve()),
+      vizPanelToPanel(this.state.panel)
+    );
+    this.setState({ isDirty: hasChanges });
   }, 250);
 
   private _handleStateChange = (event: SceneObjectStateChangedEvent) => {
-    if (!Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, 'data')) {
-      this._updateDirty();
+    if (DashboardSceneChangeTracker.isUpdatingPersistedState(event)) {
```

Comment:
I added it as static, to keep all the logic about what changes affects the persisted state in a single place